src/pages/Home.tsx

```diff
@@ -1,3 +1,71 @@
-const Home = () => <div>Home</div>;
+import useSWR from 'swr';
+import { TextField, Box, IconButton } from '@mui/material';
+import { SearchOutlined } from '@mui/icons-material';
+import { useEffect, useState } from 'react';
+
+import WeatherCard from '../components/WeatherCard';
+import { fetcher } from '../utils/fetcher';
+import { toCurrentWeatherData } from '../utils/mapper';
+
+const Home = () => {
+	const [input, setInput] = useState<string>('');
+	const [location, setLocation] = useState<string>('');
+	const [shouldFetch, setShouldFetch] = useState<boolean>(false);
+
+	const { data, error } = useSWR(
+		shouldFetch
+			? `current.json?key=${process.env.REACT_APP_API_KEY}&q=${location}&aqi=no`
+			: null,
+		fetcher
+	);
+
+	const search = () => {
+		setLocation(input);
+		setShouldFetch(true);
+	};
+
+	useEffect(() => {
+		if (error !== undefined) {
+			setShouldFetch(false);
+		}
+	}, [error]);
+
+	const handleKeyDown = (event: { key: string }) => {
+		if (event.key === 'Enter') {
+			search();
+		}
+	};
+
+	return (
+		<Box
+			sx={{
+				display: 'flex',
+				flexDirection: 'column',
+				alignItems: 'center',
+				width: '90%'
+			}}
+		>
+			<TextField
+				sx={{ width: '70%', mb: '3rem' }}
+				id="outlined-helperText"
+				label="City"
+				helperText="search by city name"
+				value={input}
+				onChange={e => {
+					setInput(e.target.value);
+				}}
+				onKeyDown={handleKeyDown}
+				InputProps={{
+					endAdornment: (
+						<IconButton onClick={search}>
+							<SearchOutlined />
+						</IconButton>
+					)
+				}}
+			/>
+			{data && <WeatherCard data={toCurrentWeatherData(data)} />}
```

Comment:
hmm, osobne bych to nemapoval az tady, budto primo v useSWR a nebo bych nechal ty data tak, jak prijdou z API (ten format dat mi prijde celkem rozumny a budeme mit vsechny informace k dispozici). Navic ten mapper mi prijde celkem zbytecny, v podstate jen presype data z jednoho objektu do noveho.